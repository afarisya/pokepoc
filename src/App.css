body {
  background-color: rgba(0,0,0,0.025) !important;
}

.App {
  text-align: center;
  padding-top: 100px;
  padding-bottom: 50px;
}

@media(max-width: 767px) {
  .App {
    padding-top: 56px;
    padding-bottom: 0;
  }
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.navbar {
  background-color: #ff1c1c !important;
  color: #fff !important;
}

.navbar a {
  color: #fff !important;
}

.card {
  box-shadow: 1px 1px 3px rgba(0,0,0,0.15);
  border: none !important;
  margin: 3px;
  /* margin-bottom: 15px; */
}

.card:hover {
  box-shadow: 2px 2px 6px rgba(0,0,0,0.15);
}

.pokemon-col {
  padding: 5px !important;
}

.pokemon-col a {
  text-decoration: none !important;
}

.pokemon-list-main-row {
  padding: 7px;
  margin-top: 10px;
  margin-bottom: 20px;
}

.card-title span {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  max-width: 100%;
}

.pokemon-nickname {
  color: #ff1c1c !important;
}

.badge {
  font-size: 14px !important;
  padding-bottom: 8px !important;
}