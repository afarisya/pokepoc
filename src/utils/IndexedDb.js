// import store from '../store'
// import { setIndexedDB } from "../reducers/GlobalReducers";
// import { setUserPicture } from "../reducers/ProfileReducers";
// import { sndAudioStream, reqInitSendBlobChange, sndEosStream, sndEndStream } from '../reducers/NewMeetingReducers';

// import defaultPicture from '../img/admin.png';


export var db;

// IndexedDB
var indexedDB = window.indexedDB || window.webkitIndexedDB || window.mozIndexedDB || window.OIndexedDB || window.msIndexedDB;
// var IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.OIDBTransaction || window.msIDBTransaction;
var dbVersion = 1.0;

// Create/open database
var request = indexedDB.open("MyPokemonList", dbVersion);

var lastIndex = null;

request.onsuccess = function (event) {
    console.log("Success creating/accessing IndexedDB database");

    db = request.result;

    db.onerror = function (event) {
        console.log("Error creating/accessing IndexedDB database");
    };
    
}

request.onerror = function (event) {
    console.log("Error creating/accessing IndexedDB database");
};

// For future use. Currently only in latest Firefox versions
request.onupgradeneeded = function (event) {
    // console.log("Upgrading IndexedDb");
    var dataBase = event.target.result;

    var store = dataBase.createObjectStore("pokemons", { 
                    keyPath: 'pokemonId'//,
                    // autoIncrement: true 
                });
    store.createIndex("pokemonId", "pokemonId", { unique: true });
    store.createIndex("pokemonName", "pokemonName", { unique: false });
    store.createIndex("pokemonNickname", "pokemonNickname", { unique: false });

};

export var createObjectStore = function (dataBase, objStoreName) {
    // Create an objectStore
    // console.log("Creating objectStore")
    dataBase.createObjectStore(objStoreName);
};

export var getMyPokemonList = function(offset, limit){
    return new Promise((resolve, reject) => {
        var objectStore = db.transaction("pokemons").objectStore("pokemons");
        var pokemons = [];

        var iterate = 0;

        var request = objectStore.openCursor();

        request.onsuccess = function(event) {
            var cursor = event.target.result;
            if (cursor) {
                if ( iterate >= offset && pokemons.length < limit ) {
                    pokemons.push(cursor.value);
                }

                console.log(pokemons)
                if ( pokemons.length === limit ) {
                    getMyPokemonCount()
                        .then((count) => {
                            resolve(
                                {
                                    count: iterate,
                                    pokemons: pokemons
                                }
                            );
                        })
                        .catch((err) => {
                            reject("error")
                        })
                }

                iterate+=1;
                cursor.continue();

            } else {
                getMyPokemonCount()
                    .then((count) => {
                        resolve(
                            {
                                count: iterate,
                                pokemons: pokemons
                            }
                        );
                    })
                    .catch((err) => {
                        reject("error")
                    })
            }
        }

        request.onerror = function(event) {
            reject("error")
        }
    })


    // IDBKeyRange.bound(0, N, false, true);
}

export var getMyPokemonCount = function(){
    return new Promise((resolve, reject) => {
        var transaction = db.transaction(['pokemons'], 'readonly');
        var objectStore = transaction.objectStore('pokemons');
            
        var countRequest = objectStore.count();

        countRequest.onsuccess = function() {
            console.log(countRequest.result)
            resolve(countRequest.result);
        }

        countRequest.onerror = function(event) {
            reject("error");
        };
    })
}

export var getPokemonById = function(id){
    return new Promise((resolve, reject) => {
        var transaction = db.transaction(["pokemons"]);
        var objectStore = transaction.objectStore("pokemons");
        var request = objectStore.get(id);
        
        request.onsuccess = function(event) {
            // Do something with the request.result!
            resolve(request.result);
        };

        request.onerror = function(event) {
            reject("error");
        };
    })
}

export var addPokemon = function(obj){    
    return new Promise((resolve, reject) => {
        var request = db.transaction(["pokemons"], "readwrite")
                    .objectStore("pokemons")
                    .add(obj);

        // transaction.objectStore(objStoreName).put(obj, key);

        request.onsuccess = function(event) {
            resolve("success");
        };

        request.onerror = function(event) {
            reject("error");
        };
    })

};

export var releasePokemon = function(pokemonId){
    return new Promise((resolve, reject) => {
        var request = db.transaction(["pokemons"], "readwrite")
                    .objectStore("pokemons")
                    .delete(pokemonId);

        request.onsuccess = function(event) {
            resolve("success");
        };

        request.onerror = function(event) {
            reject("error");
        };
    })
}