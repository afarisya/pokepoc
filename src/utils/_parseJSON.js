export function _parseJSON(response) {
    return response.text()
        .then( text => {
            return text ? JSON.parse(text) : {}
    })
}