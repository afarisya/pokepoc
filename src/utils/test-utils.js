import { ConnectedRouter } from 'react-router-redux';
import { mount } from 'enzyme';
import { mergeDeepRight } from 'ramda';
import React from 'react';
import { Provider } from 'react-redux';
import { createStoreWithMiddleWare, history } from '../store';
import rootReducer from '../reducers';

export const mockData = {
  pokemon: {
    id: 1,
    species: { name: 'bulbasaur' },
    sprites: {
      front_default:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/26.png'
    }
  }
};

export const makeMountRender = (Component, defaultProps = {}) => {
  return (customProps = {}) => {
    const props = {
      ...defaultProps,
      ...customProps
    };
    return mount(<Component {...props} />);
  };
};

export const makeStore = (customState = {}) => {
  const root = rootReducer({}, { type: '@@INIT' });
  const state = mergeDeepRight(root, customState);

  return createStoreWithMiddleWare(rootReducer, state);
};

export const reduxify = (Component, props = {}, state = {}) => {
  return function reduxWrap() {
    return (
      <Provider store={makeStore(state)}>
        <ConnectedRouter history={history}>
          <Component {...props} />
        </ConnectedRouter>
      </Provider>
    );
  };
};

export const snapshotify = reactWrapper => {
  return reactWrapper.html();
};

export const mocker = apiMock => ({
  fetchRandomPokemon() {
    apiMock
      .onGet(/https:\/\/pokeapi.co\/api\/v2\/pokemon\/\d+/)
      .reply(config => {
        const numberCheck = RegExp(/pokemon\/(\d+)/);
        const pokemonInt = Number(numberCheck.exec(config.url)[1]);

        if (pokemonInt > 151) {
          return [500];
        }

        return [200, mockData.pokemon];
      });

    return this;
  }
});

export const ticks = (callbacks = []) => {
  if (callbacks.length < 1) {
    return null;
  }

  setTimeout(() => {
    const cb = callbacks.shift();

    if (cb) {
      cb();
    }

    ticks(callbacks);
  });
};
