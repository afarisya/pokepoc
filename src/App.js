// JS Library
import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'react-router-redux';
import { Route, Switch, Redirect } from 'react-router'

import store, { history } from './store';

// Components
import PokemonListPage from './js/PokemonListPage';
import PokemonDetailPage from './js/PokemonDetailPage';
import MyPokemonListPage from './js/MyPokemonListPage';
import Header from './js/Header';

// CSS
import './App.css';

// etc
import logo from './logo.svg';


function App() {
	return (
		<Provider store={store}>
			<ConnectedRouter history={history}>
				<div className="App">
					<Header />
					<Switch>
						<Redirect exact from="/" to={"/pokemons"} />
						<Route path={"/pokemons"} component={PokemonListPage} />
						<Route path={"/pokemon/:pokemonId"} component={PokemonDetailPage} />
						<Route path={"/my-pokemons"} component={MyPokemonListPage} />
						<Redirect from="/:someroute" to="/" />
					</Switch>
				</div>
			</ConnectedRouter>
		</Provider>
	);
}

export default App;
