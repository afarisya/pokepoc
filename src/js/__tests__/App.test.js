import React from 'react';
import ReactDOM from 'react-dom';
import { makeMountRender, reduxify, snapshotify } from '../../utils/test-utils';
import App from '../../App';

describe('<App />', function() {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', function() {
    const wrapper = makeMountRender(reduxify(App))();
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });
});
