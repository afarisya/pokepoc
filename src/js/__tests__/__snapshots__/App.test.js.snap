// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `"<div class=\\"App\\"><nav class=\\"navbar navbar-expand-md navbar-light fixed-top\\"><a class=\\"navbar-brand\\"></a><button type=\\"button\\" class=\\"navbar-toggler\\"><i class=\\"material-icons\\">menu</i></button><div class=\\"collapse navbar-collapse\\"><ul class=\\"ml-auto navbar-nav\\"><li class=\\"nav-item\\"><a class=\\"nav-link active\\" style=\\"font-weight: 700;\\" aria-current=\\"page\\" href=\\"/pokemons\\">All Pokemons</a></li><li class=\\"nav-item\\"><a class=\\"nav-link\\" style=\\"font-weight: normal;\\" href=\\"/my-pokemons\\">My Pokemons</a></li></ul></div></nav><div><div class=\\"container\\"><div class=\\"pokemon-list-main-row row\\"><div role=\\"status\\" style=\\"margin: 50% auto;\\" class=\\"spinner-border text-danger\\"><span class=\\"sr-only\\">Loading...</span></div></div><div style=\\"padding: 0px 15px;\\" class=\\"row\\"><div style=\\"margin-bottom: 20px;\\"><nav class=\\"\\" aria-label=\\"Page navigation example\\"><ul class=\\"pagination\\"><li class=\\"page-item disabled\\"><button class=\\"page-link\\" aria-label=\\"First\\"><span aria-hidden=\\"true\\">«</span><span class=\\"sr-only\\">First</span></button></li><li class=\\"page-item disabled\\"><button class=\\"page-link\\" aria-label=\\"Previous\\"><span aria-hidden=\\"true\\">‹</span><span class=\\"sr-only\\">Previous</span></button></li><li class=\\"page-item disabled\\"><button class=\\"page-link\\" aria-label=\\"Next\\"><span aria-hidden=\\"true\\">›</span><span class=\\"sr-only\\">Next</span></button></li><li class=\\"page-item disabled\\"><button class=\\"page-link\\" aria-label=\\"Last\\"><span aria-hidden=\\"true\\">»</span><span class=\\"sr-only\\">Last</span></button></li></ul></nav></div></div></div></div></div>"`;
